frontend/src/lib/lemon-ui/LemonSegmentedSelect/LemonSegmentedSelect.tsx

```diff
@@ -0,0 +1,23 @@
+import { useWindowSize } from 'lib/hooks/useWindowSize'
+import React from 'react'
+
+import { LemonSegmentedButton, LemonSegmentedButtonProps } from '../LemonSegmentedButton'
+import { LemonSelect, LemonSelectProps } from '../LemonSelect'
+
+export type LemonSegmentedSelectProps<T extends React.Key> = LemonSegmentedButtonProps<T> &
+    LemonSelectProps<T> & {
+        shrinkOn: number
+    }
+
+export function LemonSegmentedSelect<T extends React.Key>({
+    shrinkOn,
+    ...props
+}: LemonSegmentedSelectProps<T>): JSX.Element {
+    const { width = 0 } = useWindowSize()
+
+    if (props.options.length >= shrinkOn || width < props.options.length * 100) {
```

Comment:
is 100 arbitrarily magical or...

Reply:
Kinda :)

It does the job just right for the current use case. Future uses should probably turn make that a customizable prop